### DISCUSSION TITLE
The method eth_getProof does not exist

### DISCUSSION THREAD

**Discussion by NorwegianGoat** [0xPolygon/polygon-edge#508] [Ideas] [answered]
### Description

As from the title, when the JSON-RPC method eth_getProof is called the result is the following:
> {'jsonrpc': '2.0', 'id': 10, 'error': {'code': -32601, 'message': 'the method eth_getProof does not exist/is not available'}}

### Your environment

* Debian 11
* Polygon-Edge 0.1.0

### Steps to reproduce

1. Deploy a contract on your Polygon-Edge chain istance.
2. Call the method eth_getProof on it (e.g. `curl -X POST --data '{"jsonrpc":"2.0","method":"eth_getProof","params":["0xAcfBd9Cbf113f1D7a10785c6C7eBf3053622f680",["0x0000000000000000000000000000000000000000000000000000000000000000"],"latest"],"id":1}' -H "Content-type:application/json" http://192.168.1.110:8545`).
The first param is the smart contract address, the second one is an array of storage keys that needs to be proofed, the third one is the block you want.

### Expected behaviour

An account object with [proofs](https://eips.ethereum.org/EIPS/eip-1186) should be returned.

### Actual behaviour

The JSON-RPC returns the error in the description section.

**Comment by zivkovicmilos** [comment, marked as answer]
Hey @NorwegianGoat,

Thank you for reaching out.

This endpoint is not officially supported by the Ethereum spec:
https://github.com/ethereum/EIPs/issues/1186

Only certain implementations support it out of the box.

We would need to discuss it internally whether or not to add support for this endpoint since we follow the spec and rarely deviate when the JSON-RPC layer is concerned.

What do you want to utilize this endpoint for?

What other endpoints would you also like to see supported?

**Reply by NorwegianGoat** [reply]
Hi, I wanted to use this endpoint to get Merkle proofs to verify them later via a smart contract. 
Since the endpoint is not supported I'm going to try other ways to reach my goal.

Currently, this is the only endpoint I need I didn't find, so it's not a big deal. Thank you for the response.

**Reply by zivkovicmilos** [reply]
Got it.

You may want to take a look at this library we worked on, you might find it interesting:
https://github.com/Trapesys/fastmerkle

**Comment by RonTuretzky** [comment]
Would really appreciate an out fo the box solution for the L1 "getEventMembershipbyBlockNumber" args, such that given an event index in a block , proof could easily be generated to be submitted to the L1  @zivkovicmilos
